user/tests/wiring/no_fixture/cellular.cpp

```diff
@@ -157,6 +157,56 @@ test(CELLULAR_07_rssi_is_valid) {
     assertMoreOrEqual(s.rssi, -150);
 }
 
+test(CELLULAR_08_sigstr_is_valid) {     // collects signal strength and quality
+    connect_to_cloud(6*60*1000);
+    assertTrue(Particle.connected());
+    CellularSignal s;
+    for (int x = 0; x < 10; x++) {
+        s = Cellular.RSSI();
+        if (s.getStrengthValue() <= 0 && s.getQualityValue() <= 0) {
+            break;
+        }
+        delay(5000);
+    }
+
+    // Verify RAT
+    uint8_t rat = s.getAccessTechnology();
+
+    assertNotEqual(rat, NET_ACCESS_TECHNOLOGY_UNKNOWN || NET_ACCESS_TECHNOLOGY_NONE);
+
+    // Verify strength and quality values
+    switch (rat) {
+        case NET_ACCESS_TECHNOLOGY_GSM:     // GSM strength [-111, -48] and quality [?, ?]
+            assertLessOrEqual(s.getStrengthValue(), -48);
+            assertMoreOrEqual(s.getStrengthValue(), -111);
+            assertLessOrEqual(s.getQualityValue(), 0);
+            assertMoreOrEqual(s.getQualityValue(), -20);
```

Comment:
Wanting to confirm the limits on GSM and EDGE might not be correct. Is the correct qualityvalue range between [0,7]?

Reply:
https://github.com/particle-iot/device-os/blob/develop/hal/network/ncp/cellular/cellular_hal.cpp#L395

GSM: 3GPP TS 45.008 8.2.4 [0.14%, 18.10%]

EDGE: 3GPP TS 45.008 10.2.3.3 [-0.60, -3.70]

Reply:
Added!